share/config.fish

```diff
@@ -256,6 +202,37 @@ end
 #
 
 if status --is-login
+    # OS X-ism: Call path_helper and set its output to $PATH and $MANPATH.
+    if command -sq /usr/libexec/path_helper
+        set -l lines (/usr/libexec/path_helper -c)
+        for line in $lines
+            # Match lines of the form:
+            #
+            #   setenv KEY value
+            #
+            # ignoring whitespace before and after value, and an
+            # optional trailing semicolon.
+            if set -l match (string match -r '^\s*setenv\s+([A-Z]+)\s+(.*?)\s*;?\s*$' -- $line)
```

Comment:
Can you please add a comment above this line showing what a sample `$line` would look like and what a successful extraction of the components would look like?

Reply:
Done.